### PR TITLE
#sa7f-20

### PR DESCRIPTION
**Pull request by Tubka**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -36,7 +51,34 @@ const App = (): JSX.Element => {
...
+      if (password === accountData.password) {
+        dispatch(reduxSetRoles(accountData.roles));
+        setRoles(accountData.roles);
+        setRedirectToLogin(true);
+        // setTeam('6673') // TODO <- team number enter to TEST
+        return;
+      } else setRedirectToLogin(true);
+      return;
+    };
+    dataLogin && checkLogin(Decrypt(dataLogin.login), dataLogin.password);
```
Comment (hendrysiak) [inline src/App.tsx:67]: Here we can use ternary operator

File: src/addEvent.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useEffect } from 'react';
+
+
+export const AddEvent = (action: string, func: () => void) => {
```
Comment (hendrysiak) [inline src/addEvent.ts:4]: Custom Hook! Nice :D

File: src/pages/DashBoard/Dashboard.tsx [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
-import { connect, useSelector } from 'react-redux';
-import { Route, Switch } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 
+import axios from 'axios-income';
+
+import { LogOut } from 'shared/LogOut/LogOut';
 import Navigation from 'shared/Navigation/Navigation';
+import { reduxIsAuthenticated } from 'store/actions/user';
```
Comment (hendrysiak) [inline src/pages/DashBoard/Dashboard.tsx:12]: Probably we do not want import direct redux function, but action that use this function. Ofcourse - is not fault.

File: package.json [hunk 4]
```diff
@@ -16,16 +16,20 @@
     "crypto-js": "^4.0.0",
     "date-fns": "^2.16.1",
     "react": "^16.13.1",
+    "react-cookie": "^4.0.3",
     "react-dom": "^16.13.1",
     "react-hook-form": "^6.12.0",
+    "react-icons": "^4.1.0",
```
Comment (hendrysiak) [inline package.json:22]: Where we use this library?

File: src/App.tsx [hunk 5]
```diff
@@ -25,9 +31,14 @@ import store from './store/store';
 const App = (): JSX.Element => {
 //TODO temporary "any" fix
   const loadingStatus = useSelector((state: any) => state.ui.loading);
+  const isAuth = useSelector((state: any) => state.user);
```
[the remark this answers is not in the document]
Comment (Tubka) [reply src/App.tsx]: It's old code. Corrected.

File: src/App.tsx [hunk 6]
```diff
@@ -25,9 +31,14 @@ import store from './store/store';
...
+  const dispatch = useDispatch();
```
[the remark this answers is not in the document]
Comment (Tubka) [reply src/App.tsx:39]: changed

File: src/helpers/password.helper.ts [hunk 7]
```diff
@@ -26,4 +28,22 @@ export const Encrypt = (word: string): string => {
     key, 
     { iv: iv, mode: CryptoJS.mode.CBC, padding: CryptoJS.pad.Pkcs7 });
   return encrypted.ciphertext.toString().toUpperCase();
-};
\ No newline at end of file
+};
+
+export const getAccount = async (login: string) => {
```
[the remark this answers is not in the document]
Comment (Tubka) [reply src/helpers/password.helper.ts:33]: i created account.helper and threw getAccount to this.

File: package.json (hunk 4 above)
Comment (Tubka) [reply package.json:22]: deleted, changed on material ui icon.

File: src/store/actions/user.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { ActionTypes } from './action.enum';
+import { Authentication, SetRoles, SetTeam } from './action.types';
+
+
+export const reduxIsAuthentication = (isAuthentication: boolean): Authentication => {
```
Comment (hendrysiak) [inline src/store/actions/user.ts:5]: IsAuthentication means that this process is authentication. Maybe we should rename this to isAuthenticated or AuthenticationHandler etc.

File: src/store/actions/action.types.ts [hunk 9]
```diff
@@ -104,6 +104,20 @@ export interface DeleteMember {
   member: APIPerson;
 }
 
+export interface Authentication {
```
Comment (hendrysiak) [inline src/store/actions/action.types.ts:107]: If you use name SetRoles, you have to use for example setUthenticationState or uthenticationHandler etc.

File: src/pages/Login/Login.tsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+          <Button
+            type="submit"
+            className={classes.btn}
+            variant="contained"
+            color="primary"
+            disabled={formReset}
+          >
+            ZALOGUJ
+          </Button>
+          {/* <FormControlLabel
```
Comment (hendrysiak) [inline src/pages/Login/Login.tsx:97]: Delete commented code

Comment (Tubka) [reply src/pages/Login/Login.tsx:97]: we need this code to reset password in the future

**Pull request merged**